_test/general.test.php

```diff
@@ -58,4 +78,408 @@ public function test_plugin_conf() {
+     * @expectedException InvalidArgumentException
+     */
+    public function test_call_export_with_start_value_below_zero() {
+        $pageId = 'demopage6';
+        saveWikiText($pageId, self::PAGE_CONTENT_PLACEHOLDER, '');
+        idx_addPage($pageId, '', '');
+
+        $conf = array();
+        $DokuwikiXMLExport = new DokuwikiXMLExport($conf);
+        $xml = new SimpleXMLElement($DokuwikiXMLExport->generateXMLExport(-1, 0));
+    }
+
+    /**
+     * Test to ensure that parameters are correctly handled in the export call.
+     * start = 0, count = 10, total = 9;
+     */
+    public function test_call_export_with_count_greater_than_total() {
+        $pageId = array();
+        for ($i=1; $i<=9; $i++){
+            $pageId[$i] = 'demopage0' . $i;
+            saveWikiText($pageId[$i], self::PAGE_CONTENT_PLACEHOLDER, '');
+            idx_addPage($pageId[$i], '', '');
+        }
+
+        $conf = array();
+        $DokuwikiXMLExport = new DokuwikiXMLExport($conf);
+        $xml = new SimpleXMLElement($DokuwikiXMLExport->generateXMLExport(0, 10));
+
+        $start = implode('', ($xml->xpath('/findologic/items/@start')));
+        $count = implode('', ($xml->xpath('/findologic/items/@count')));
+        $total = implode('', ($xml->xpath('/findologic/items/@total')));
+
+        $expectedStart = 0;
+        $expectedCount = $expectedTotal = 9;
+
+        $this->assertEquals($expectedStart, $start, 'Expected start value should match "0" when export is called with start "0".');
+        $this->assertEquals($expectedCount, $count, 'Expected count value should match "9" when export is called with "10" and $total is "9".');
+        $this->assertEquals($expectedTotal, $total, 'Expected total value should match "9" when DokuWiki has 9 pages.');
+    }
+
+    //======================================================================
+    // PLUGIN SETTING TESTS
+    //======================================================================
+
+    /**
+     * Test to ensure that the configuration works as expected and excluding of pages work properly.
+     * Correct page needs to be exported.
+     */
+    public function test_plugin_setting_exclude_pages() {
+        $pageIds = array('settingtest1337', 'excludeme1', 'excludeme2');
+        foreach ($pageIds as $pageId) {
+            saveWikiText($pageId, self::PAGE_CONTENT_PLACEHOLDER, '');
+            idx_addPage($pageId, '', '');
+        }
+
+        $conf = array();
+        $conf['plugin']['findologicxmlexport']['excludePages'] = 'excludeme1, excludeme2';
+
+        $DokuwikiXMLExport = new DokuwikiXMLExport($conf);
+        $xml = new SimpleXMLElement($DokuwikiXMLExport->generateXMLExport(0, 20));
+
+        $total = implode('', ($xml->xpath('/findologic/items/@total')));
+
+        $ordernumbers = $xml->xpath('/findologic/items/item/allOrdernumbers/ordernumbers/ordernumber');
+        $ordernumber = $ordernumbers[0];
+
+        $expectedTotal = 1;
```

Comment:
You should also check that there is only one `/findologic/items/item` in addition to checking `total`.

Reply:
I check if a second item exists.
No &rarr; correct. :ballot_box_with_check: 
Yes &rarr; error. :x: